provider/packet/environ_network.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package packet
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/context"
+	"github.com/juju/names/v4"
+)
+
+var _ environs.Networking = (*environ)(nil)
+
+// Subnets returns basic information about subnets known by the provider for
+// the environment.
+func (e *environ) Subnets(ctx context.ProviderCallContext, inst instance.Id, subnetIDs []network.Id) ([]network.SubnetInfo, error) {
+	subnets := []network.SubnetInfo{}
+
+	if inst != instance.UnknownId {
+		device, _, err := e.packetClient.Devices.Get(string(inst), nil)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		for _, n := range device.Network {
+			subnetID, cidr, err := makeSubnetIDForNetwork("device-"+device.Hostname, n.Address, strconv.Itoa(n.CIDR))
```

Comment:
@achilleasa I've played with device and hostname but i'm not sure what should go here either

Reply:
Since multiple machines (devices) will be sharing one or more subnets, the subnet ID should be unique _per subnet_ and **not** per machine. More specifically, juju will first call `Subnets()` with `instance.UnknownId` expecting the provider to return the list of subnets known to the environment. 

The call with a particular instance ID will only be used when querying the subnets that a particular instance is connected to, i.e. a subset of the full set of subnets known by the provider.

Reply:
so for the networkname i should use actual subnet?